### PR TITLE
Use templates in legend items

### PR DESCRIPTION
**Pull request by roman-simionov**
Now, you can use custom SVG in legend markers.

custom_marker

```html
<dx-chart
title="Noisy and Original Signals"
[dataSource]="data">
  <dxo-common-series-settings argumentField="argument">
    <dxo-point [visible]="false"></dxo-point>
  </dxo-common-series-settings>
  <dxi-series name="Noisy Signal" valueField="value" tag="M 18,8 L 12,12 L 7,3 L 0,7.4 L 0,10 L 6,6 L 11,15 L 18,10.6 ">
  </dxi-series>
  <dxi-series name="Original signal" valueField="originalValue"
    tag="M18,8c-0.4,0.4-1.3,1.1-2.4,1.5c-2.1,0.8-4.1,0.2-5.9-1.8C6.2,3.9,2.4,4.8,0,6.3V9c0.4-0.4,1.3-1.1,2.4-1.5  c2.1-0.8,4.1-0.2,5.9,1.8c1.9,2,3.8,2.7,5.5,2.7c1.6,0,3-0.6,4.2-1.3V8z">
  </dxi-series>
  <dxo-legend markerTemplate="myTemplate" >
    <svg *dxTemplate="let item of 'myTemplate'" attr.fill="{{item.markerAttributes.fill}}">
        <rect x="0" y="0" width="18" height="18" opacity="0.3"></rect>
        <path attr.d="{{item.series.tag}}"></path>
    </svg>
  </dxo-legend>
</dx-chart>
```

### CODE REVIEW

**Comment by alexanderPolosatov** [comment]
LGTM

File: ts/dx.all.d.ts [hunk 1]
```diff
@@ -6193,8 +6194,6 @@ declare module DevExpress.viz {
         points?: Array<piePointObject>;
         /** @name PieChartLegendItem.text */
         text?: any;
```
Comment (IlyaKhD) [inline ts/dx.all.d.ts:6196]: Shouldn't we change the `text` prop type to `string`?

File: ts/dx.all.d.ts [hunk 2]
```diff
@@ -5901,6 +5895,17 @@ declare module DevExpress.viz {
         /** @name BaseLegend.visible */
         visible?: boolean;
     }
+    /** @name BaseLegendItem */
+    export interface BaseLegendItem {
```
Comment (IlyaKhD) [inline ts/dx.all.d.ts:5899]: Just to make sure: should all the derived types (`BarGaugeLegendItem`,`BaseChartLegendItem`, `FunnelLegendItem `, etc.) have all these options (`markerAttributes`, `size`, `text`, `visible`)?

Comment (roman-simionov) [reply ts/dx.all.d.ts:5899]: Yes, these properties should be in all LegendItems

File: ts/dx.all.d.ts (hunk 1 above)
Comment (roman-simionov) [reply ts/dx.all.d.ts:6196]: This property filled from point argument and can be non-string value

**Review by IlyaKhD — APPROVED** [review]

File: ts/dx.all.d.ts (hunk 1 above)
Comment (IlyaKhD) [reply ts/dx.all.d.ts:6196]: 👌

File: ts/dx.all.d.ts (hunk 2 above)
Comment (IlyaKhD) [reply ts/dx.all.d.ts:5899]: 👌

**Pull request merged**